clash-ghc/src-ghc/Clash/GHC/LoadModules.hs

```diff
@@ -429,16 +441,89 @@ loadModules useColor hdl modName dflagsM idirs = do
     let annExtDiff = reportTimeDiff annTime extTime
     MonadUtils.liftIO $ putStrLn $ "GHC: Parsing annotations took: " ++ annExtDiff
 
+    let famInstEnvs' = (fst famInstEnvs, modFamInstEnvs)
+        allTCInsts   = FamInstEnv.famInstEnvElts (fst famInstEnvs')
+                         ++ FamInstEnv.famInstEnvElts (snd famInstEnvs')
+
+        knownConfs   = filter (\x -> "KnownConf" == nameString (FamInstEnv.fi_fam x)) allTCInsts
+
+#if MIN_VERSION_ghc(8,10,0)
+        fsToText     = Text.decodeUtf8 . FastString.bytesFS
+#else
+        fsToText     = Text.decodeUtf8 . FastString.fastStringToByteString
+#endif
+
+        famToDomain  = fromMaybe (error "KnownConf: Expected Symbol at LHS of type family")
+                         . fmap fsToText . Type.isStrLitTy . head . FamInstEnv.fi_tys
+        famToConf    = unpackKnownConf . FamInstEnv.fi_rhs
+
+        knownConfNms = fmap famToDomain knownConfs
+        knownConfDs  = fmap famToConf knownConfs
+
+        knownConfMap = HashMap.fromList (zip knownConfNms knownConfDs)
+
     return ( allBinders
            , lbClassOps
            , lbUnlocatable
-           , (fst famInstEnvs, modFamInstEnvs)
+           , famInstEnvs'
            , topEntities'
            , lbPrims
            , reprs1
            , primGuards
+           , knownConfMap
            )
 
+-- | Given a type that represents the RHS of a KnownConf type family instance,
+-- unpack the fields of the DomainConfguration and make a VDomainConfiguration.
+--
+unpackKnownConf :: Type.Type -> VDomainConfiguration
+unpackKnownConf ty
+  | [d,p,ae,rk,ib,rp] <- Type.tyConAppArgs ty
+    -- Domain name
+  , Just dom <- fmap FastString.unpackFS (Type.isStrLitTy d)
+    -- Period
+  , Just period <- fmap naturalFromInteger (Type.isNumLitTy p)
+    -- Active Edge
+  , aeTc <- Type.tyConAppTyCon ae
+  , Just aeDc <- TyCon.isPromotedDataCon_maybe aeTc
+  , aeNm <- OccName.occNameString $ Name.nameOccName (DataCon.dataConName aeDc)
+    -- Reset Kind
+  , rkTc <- Type.tyConAppTyCon rk
+  , Just rkDc <- TyCon.isPromotedDataCon_maybe rkTc
+  , rkNm <- OccName.occNameString $ Name.nameOccName (DataCon.dataConName rkDc)
+    -- Init Behaviour
+  , ibTc <- Type.tyConAppTyCon ib
+  , Just ibDc <- TyCon.isPromotedDataCon_maybe ibTc
+  , ibNm <- OccName.occNameString $ Name.nameOccName (DataCon.dataConName ibDc)
+    -- Reset Polarity
+  , rpTc <- Type.tyConAppTyCon rp
+  , Just rpDc <- TyCon.isPromotedDataCon_maybe rpTc
+  , rpNm <- OccName.occNameString $ Name.nameOccName (DataCon.dataConName rpDc)
+  = VDomainConfiguration dom period
+      (asActiveEdge aeNm)
+      (asResetKind rkNm)
+      (asInitBehaviour ibNm)
+      (asResetPolarity rpNm)
+
+  | otherwise
+  = error $ $(curLoc) ++ "Could not unpack domain configuration."
+ where
+  asActiveEdge "Rising"  = Rising
+  asActiveEdge "Falling" = Falling
+  asActiveEdge x = error $ $(curLoc) ++ "Unknown active edge: " ++ show x
```

Comment:
`read`?

Reply:
No `Read` instances, and the error from `Read` is quite obtuse (`*** Exception: Prelude.read: no parse`)

Reply:
deriving Read + readMaybe? :)

Reply:
You've got me there, I'll make a small PR for it tomorrow :+1: